### PR TITLE
Changed code to improve output for files under test/e2e/instrumentation

### PR DESCRIPTION
**Pull request by NikhilSharmaWe**
#### What type of PR is this?
Enhancement

#### What this PR does / why we need it:
For better (more informative) output for developers when test fails. Changed files are under test/e2e/instrumentation for this PR.
#### Which issue(s) this PR fixes:

Part of #105678

#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by NikhilSharmaWe** [comment]
@pohly could you please review the changes made.

File: test/e2e/instrumentation/core_events.go [hunk 1]
```diff
@@ -84,7 +84,9 @@ var _ = common.SIGDescribe("Events", func() {
 				break
 			}
 		}
-		framework.ExpectEqual(foundCreatedEvent, true, "unable to find the test event")
+		if !foundCreatedEvent{
+			framework.Failf("unable to find test event with test name : %v and namespace name : %v in %+v", eventTestName, f.Namespace.Name, eventsList.Items)
+		}
```
Comment (pohly) [inline test/e2e/instrumentation/core_events.go:89]: See the other PR about formatting with ` : `. Also check the Go formatting.

**Review by pohly — CHANGES_REQUESTED** [review]

**Comment by NikhilSharmaWe** [comment]
@pohly made changes according to your review.

File: test/e2e/instrumentation/core_events.go (hunk 1 above)
Comment (pohly) [reply test/e2e/instrumentation/core_events.go:89]: `<event name>/<namespace>` doesn't make sense.

File: test/e2e/instrumentation/core_events.go [hunk 2]
```diff
@@ -121,7 +123,9 @@ var _ = common.SIGDescribe("Events", func() {
 				break
 			}
 		}
-		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
+		if foundCreatedEvent {
+			framework.Failf("should not have found test event %s/%s in %+v", eventTestName, f.Namespace.Name, eventsList.Items)
```
Comment (pohly) [inline test/e2e/instrumentation/core_events.go:127]: Same here and below.

**Review by pohly — CHANGES_REQUESTED** [review]

File: test/e2e/instrumentation/core_events.go (hunk 2 above)
Comment (NikhilSharmaWe) [reply test/e2e/instrumentation/core_events.go:127]: So should I include name and namespace in the output. If yes should I do like this -
`if !foundCreatedEvent{
         framework.Failf("unable to find test event with test name : %v and namespace name : %v in %+v", eventTestName, f.Namespace.Name, eventsList.Items)
}`

Comment (pohly) [reply test/e2e/instrumentation/core_events.go:127]: ` : ` as separator looks ugly. Messages should be plain English, not some made up markup.

```suggestion
			framework.Failf("Should not have found test event %s in namespace %s, full list of events is %+v", eventTestName, f.Namespace.Name, eventsList.Items)
```

**Review by NikhilSharmaWe — COMMENTED** [review]
@pohly what will represent the list of events with cluster scope and other with namespace scope. In this case it can be used for the output https://github.com/kubernetes/kubernetes/pull/106233/files#diff-a80c889c0b731cd66a65628d6e7b900992b605ffa73f25c6d9dd71f70633a063R105

**Comment by NikhilSharmaWe** [comment]
@pohly could you please review the changes made.

**Comment by pohly** [comment]
Format string and args don't match:
```
test/e2e/instrumentation/core_events.go:88:4: Failf format %+v reads arg #4, but call has 3 args
```

File: test/e2e/instrumentation/events.go [hunk 3]
```diff
@@ -172,11 +176,15 @@ var _ = common.SIGDescribe("Events API", func() {
...
 		foundCreatedEvent = eventExistsInList(clientAllNamespaces, f.Namespace.Name, eventName)
-		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
+		if foundCreatedEvent {
+			framework.Failf("Should not have found test event %s in namespace %s, in list with cluster scope after deletion", eventName, f.Namespace.Name)
+		}
 
 		ginkgo.By("listing events in test namespace")
 		foundCreatedEvent = eventExistsInList(client, f.Namespace.Name, eventName)
-		framework.ExpectEqual(foundCreatedEvent, false, "should not have found test event after deletion")
+		if !foundCreatedEvent {
```
Comment (pohly) [inline test/e2e/instrumentation/events.go:185]: Here the logic of the check got inverted, hence the test failures.

Comment (NikhilSharmaWe) [reply test/e2e/instrumentation/events.go:185]: Got it

**Review by pohly — APPROVED** [review]
/lgtm
/retest

Unit test failure seems unrelated.

**Comment by NikhilSharmaWe** [comment]
@pohly I think PR is ready.

**Review by pohly — APPROVED** [review]
/lgtm

BTW, this current pattern of calling `eventExistsInList` and then checking the return code is still sub-optimal. A much better approach would have been to write a Gomega matcher because that then can produce much better failure messages.

This goes beyond the scope of this PR, though.

**Review by ehashman — COMMENTED** [review]
/triage accepted
/priority backlog
/approve

**Pull request merged**